### ISSUE TITLE
General "Website URL" confusion, affiliate registration enhancements

### ISSUE THREAD

**Issue by sumobi** [AffiliateWP/AffiliateWP#1512]
Currently, a user who is submitting the affiliate registration form can enter their `website URL`. This URL is then stored with their WordPress user profile. 

The "Website URL" field is confusing. Some random thoughts:

1. This URL entered may not necessarily be the website URL they intend to promote the site owner's products/services from. The affiliate could just enter a personal URL.

2. The affiliate registration form isn't clear on what this field is for, it just says "Website URL". Is it the site where the affiliate's intend to promote from (which is why we added it) or is it their personal URL?

3. The current website URL field is required. Some affiliates don't have websites and promote "offline". Related: https://github.com/AffiliateWP/AffiliateWP/issues/931

4. We store the website URL field in the WP user profile. WP lists this URL under "contact info". If we're aiming for this to be the site that affiliates promote the site owner's products/services from, then it can't be an affiliate's "contact" URL, which they can change at any time.

5. Affiliates can have more than 1 website URL. If the aim of the affiliate registration is to list the sites they will promote from, including social media profiles, they should be able to enter more than 1 URL. This is also extremely important when affiliate approval is enabled. The idea when reviewing an affiliate is to look at the sites/social media profiles an affiliate has submitted, and approve/reject their application based on this information.

6. Since 1.8.7 (Related: https://github.com/AffiliateWP/AffiliateWP/issues/1466) the admin can now see the URL on the affiliate edit screen. Again, this is deemed a "contact URL" by WordPress. The admin should see real sites where the affiliate is promoting from. The contact URL is also something that is shown in comments and other areas of a site. An admin shouldn't really update an affiliate's contact URL since it would change in these places.

## Proposal/s:

1. We don't use the website URL stored against the user's WP profile. We should be storing URLs in the affiliate meta DB table. We could write an upgrade routine to move the currently added website URL over to affiliate meta.

2. We allow multiple URLs to be submitted from the affiliate registration form. This could be an admin option, e.g. "Number of URLs an affiliate can submit". If more than 1 is allowed, the form would allow a repeatable field. The affiliate can then enter the website/s they intend to promote from, their social profiles (twitter, facebook etc). By providing more information to the admin, they have a better chance of being approved if affiliate approval is enabled.

3. We provide an option to make the website URL field not required (https://github.com/AffiliateWP/AffiliateWP/issues/931).

4. We provide a better label on what the "website URL" field is on the affiliate registration form. Since these are the sites the affiliate will use to promote the site owner's products/services, what do you think is a better label?

5. On the affiliate review screen (when approval is enabled) these sites will be listed.

6. On the affiliate edit screen these sites will be listed, and will be editable by the admin.

**Comment by pippinsplugins** [comment]
I agree with all of this, but I don't think we should put this in 1.9, simply because we're running out of time and 1.9 keeps growing instead of shrinking.

I propose either soon after 1.9 in a point release (even though it's a pretty major change) or 2.0.

**Comment by michaelbeil** [comment]
I would say this should be shortly after 1.9 as Pippin mentioned, but this is going to be helpful to users.

**Comment by sumobi** [comment]
Sounds good to me

**Comment by sumobi** [comment]
I vote we punt to 2.0

**Comment by pippinsplugins** [comment]
:+1:

**Comment by sumobi** [comment]
I don't think we'll have enough time for this to go in 2.0. Proposal point 3 above is done in https://github.com/AffiliateWP/AffiliateWP/issues/1443

I propose we bump it.

**Comment by pippinsplugins** [comment]
Bumped.
